mantle/cmd/kola/devshell.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Run qemu as a development shell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/coreos/mantle/util"
+	"github.com/pkg/errors"
+
+	v3 "github.com/coreos/ignition/v2/config/v3_0"
+	v3types "github.com/coreos/ignition/v2/config/v3_0/types"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/platform"
+)
+
+const devshellHostname = "cosa-devsh"
+
+func devshellSSH(configPath, keyPath string, silent bool, args ...string) exec.Cmd {
+	sshArgs := append([]string{"-i", keyPath, "-F", configPath, devshellHostname}, args...)
+	sshCmd := exec.Command("ssh", sshArgs...)
+	if !silent {
+		sshCmd.Stdin = os.Stdin
+		sshCmd.Stdout = os.Stdout
+		sshCmd.Stderr = os.Stderr
+	}
+	sshCmd.SysProcAttr = &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGTERM,
+	}
+
+	return *sshCmd
+}
+
+func readTrimmedLine(r *bufio.Reader) (string, error) {
+	l, err := r.ReadString('\n')
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(l), nil
+}
+
+func runDevShell(builder *platform.QemuBuilder, conf *v3types.Config) error {
+	tmpd, err := ioutil.TempDir("", "kola-devshell")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpd)
+
+	sshKeyPath := filepath.Join(tmpd, "ssh.key")
+	sshPubKeyPath := sshKeyPath + ".pub"
+	err = exec.Command("ssh-keygen", "-N", "", "-t", "ed25519", "-f", sshKeyPath).Run()
```

Comment:
We don't need to do this for `--devshell-console`, right?

Reply:
Right...I thought this was going to be a messy fix but in fact we can just only do this whole function in the SSH case.